resources/Microsoft.Windows.Setting.Time/Microsoft.Windows.Setting.Time.psm1

```diff
@@ -39,57 +40,107 @@ function Get-ValidTimeZone {
 
     return $timeZoneId
 }
+
+function Test-LocationSettingPermission {
+    # On Windows 11, the HKLM is the location services, whereas HKCU is the Let apps access your location setting
+    $registryKeys = @('HKLM:\Software\Microsoft\Windows\CurrentVersion\CapabilityAccessManager\ConsentStore\location', 'HKCU:\Software\Microsoft\Windows\CurrentVersion\CapabilityAccessManager\ConsentStore\location', 'HKCU:\Software\Microsoft\Windows\CurrentVersion\CapabilityAccessManager\ConsentStore\location\windows.immersivecontrolpanel*')
+    foreach ($key in $registryKeys) {
+        $res = Get-ItemProperty -Path $key -Name 'Value' -ErrorAction SilentlyContinue
+        if ($res.Value -ne 'Allow') {
+            # TODO: Or should we throw an error?
+            return $false
+        }
+    }
+
+    return $true
```

Comment:
I feel that this should be its own separate DSC resource for just Location. I'm sure there are other settings that will depend on this, not just Time/TimeZone.

Reply:
Partially, as it affects the TimeZone setting, what's your suggestion? A custom module to inherit those functions?